scripts/extensions/sams/src/components/sets/setListPanel.tsx

```diff
@@ -0,0 +1,98 @@
+// External Modules
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {Dispatch} from 'redux';
+
+// Types
+import {ISuperdesk} from 'superdesk-api';
+import {ISetItem, IStorageDestinationItem} from '../../interfaces';
+import {IApplicationState} from '../../store';
+
+// Redux Actions & Selectors
+import {confirmBeforeDeletingSet, editSet, previewSet} from '../../store/sets/actions';
+import {getSetsGroupedByState, getSelectedSetId} from '../../store/sets/selectors';
+import {getStorageDestinationsById} from '../../store/storageDestinations/selectors';
+
+// UI
+import {SetListGroup} from './setListGroup';
+
+interface IProps {
+    storageDestinations: Dictionary<string, IStorageDestinationItem>;
+
+    previewSet(set: ISetItem): void;
+    deleteSet(set: ISetItem): void;
+    editSet(set: ISetItem): void;
+
+    currentSetId?: string;
+    sets: ISetArrays;
+}
+
+type ISetArrays = {
+    draft: Array<ISetItem>;
+    usable: Array<ISetItem>;
+    disabled: Array<ISetItem>;
+};
+
+export function getSetListPanel(superdesk: ISuperdesk) {
+    const {gettext} = superdesk.localization;
+
+    const mapStateToProps = (state: IApplicationState) => ({
+        sets: getSetsGroupedByState(state),
+        storageDestinations: getStorageDestinationsById(state),
+        currentSetId: getSelectedSetId(state),
+    });
+
+    const mapDispatchToProps = (dispatch: Dispatch) => ({
```

Comment:
Can you remove instances of `mapDispatchToProps`?

Reply:
Using `mapDispatchToProps` makes the view component itself more declarative as it does not need to know about `dispatch` and the usage thereof, instead it only has to worry about what prop to execute. https://react-redux.js.org/using-react-redux/connect-mapdispatch#more-declarative

Reply:
The usage of `mapDispatchToProps` is the convention being used in this extension

Reply:
I thought we agreed on a compromise that if we are to use redux, we would minimize the boilerplate, no?

Convention or not, we should be able to justify the usage of any pattern and drop everything that doesn't provide significant value.

I think I refuted the argument about it being more declarative. Are there other reasons you think it's worth using? As I mentioned before, the reason I don't want it used, because it's an abstraction which doesn't help in any way, unless you want to connect the same component twice, which we are not doing.

I don't think we should have any different conventions in core/extensions since it's the same team maintaining both.

Reply:
I spent an entire day rewriting the redux integration to cater for your needs to reduce boilerplate, which was ultimately rejected.

You have the opinion that you don't want `mapDispatchToProps`, I have the opinion (and so does the Redux team) that it is more declarative if the `dispatch` function is not used inside the React component. I believe that any usage of Redux directly should not be used inside any React Component, that is what `mapStateToProps` and `mapDispatchToProps` is for. Using `mapDispatchToProps` does not add much boilerplate at all (in this example it is only 6 lines).

Reply:
> I spent an entire day rewriting the redux integration to cater for your needs to reduce boilerplate, which was ultimately rejected.

Thank you for doing that and am sorry if I wasn't clear enough. I only asked to drop action creator functions and inline the actions, not merge actions and reducers into one system.

Regarding `mapDispatchToProps`, it's not only about lines, but also about call stack and having to jump around the code more to discover what the code does.

For example, on button click a prop is called, I then have to jump to that prop to see what it does - I discover that it does nothing, but is just a proxy for dispatching an action. I then go to action definition to see its name and search for a reducer to finally find the actual state changes done by clicking that button. If I can skip one layer and see the actual state change quicker that's already more convenient.

> I believe that any usage of Redux directly should not be used inside any React Component

And I believe that everyone should be able to justify technical decisions rationally and explain how it helps or harms code quality.

Anyway, let's not spend too much time on this, keep the mapping and I will do an extra jump in the call stack :)